### DISCUSSION TITLE
Support log.level field by default

### DISCUSSION THREAD

**Discussion by cdalexndr** [grafana/grafana#39356] [Observability]
Elasticsearch filebeat uses log.level field to store the log level.
Elasticsearch is widely used.
Shouldn't log.level be checked by default before fallback to unknown log level?
It would make Grafana more user friendly, by supporting a common framework.

https://github.com/grafana/grafana/issues/25678

**Comment by lewinskimaciej** [comment]
This is an old discussion but we just encountered this issue now, using .NET with Serilog logging to Elasticsearch and Grafana. If we use official TextFormatter from https://github.com/elastic/ecs-dotnet to have properly formatted events then it puts the log level into 'log.level', which causes Grafana to display all logs as 'unknown' level. There should be either a fallback value to check for this as Elastic is one of the most popular data sources for Grafana) or a way to customize this setting.

**Comment by gabor** [comment]
the log-level field is now configurable in the elasticsearch datasource. @cdalexndr , @lewinskimaciej , does this solve the problem described here?

**Reply by echatman-ias** [reply]
I have this same issue and configuring it in the datasource doesn't help - different applications put the log level in different fields (I am aware of "level", "log.level", and "dissect.level" so far but we may have others too). I need this to be configurable on the panel itself.

**Reply by echatman-ias** [reply]
I also tried using "Organize fields" to rename my "log.level" field to "level" but that gets completely ignored.

**Reply by gabor** [reply]
hi @echatman-ias , thanks for the feedback! though this seems to be a different topic (make the log-level configurable in the panel itself). could you please create a separate discussion for it?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
